app/models/submission.rb

```diff
@@ -1,4 +1,30 @@
 class Submission < ApplicationRecord
+  validate :vault_balance
   belongs_to :user
   belongs_to :form
+  has_many_attached :images do |attachable|
+    attachable.variant :thumb, resize_to_limit: [100, 100]
+    enum status: [:approved, :pending, :rejected]
+  end
+
+  def vault_balance
+    submissions = user.submissions.where(form_id: Form.find_by(_type: "medical").id)
+    balance = user.roles.first.medical_allowance - (user.total_generator(submissions) + submission_total(self))
+    if balance < 0 && Form.find(self.form_id)._type == "medical"
+      errors.add(:balance, "low balance")
+    end
```

Comment:
very bad way to check validation you need to perform all these calculation if the from is for medical.... you are first making all calculations and then applying check for errors